src/lib/containers/download_list/DownloadListTable.tsx

```diff
@@ -204,28 +218,99 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
   }
 
+  const sortColumn = async (column: string) => {
+    try {
+      setIsLoading(true)
+
+      setColumns({
+        file: column === 'file' ? !columns[column] : false,
+        createdBy: column === 'createdBy' ? !columns[column] : false,
+        createdOn: column === 'createdOn' ? !columns[column] : false,
+        size: column === 'size' ? !columns[column] : false,
+      })
+
+      const filesToDownload = downloadList?.filesToDownload ?? []
+
+      filesToDownload.sort((itemA, itemB) => {
+        return sortDownLoadList(itemA, itemB, column)
+      })
+      setData({
+        ...data,
+        downloadList,
+      })
+      listUpdatedCallback?.()
+    } catch (err) {
+      console.error(err)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
   const sortDownLoadList = (
     itemA: FileHandleAssociation,
     itemB: FileHandleAssociation,
+    column: string,
```

Comment:
also need to pass in `isAscending` to the function so the sort can switch directions